Sming/SmingCore/Network/Ssl/SslFingerprints.h

```diff
@@ -14,15 +14,107 @@ enum SslFingerprintType {
 					   //    Only when the private key used to generate the certificate is used then that fingerprint
 };
 
-typedef struct {
+/** @brief Contains SSL fingerprint data
+ *  @note Lifetime as follows:
+ *  	- Constructed by application, using appropriate setXXX method;
+ *  	- Passed into HttpRequest by application, using pinCertificate method - request is then queued;
+ *  	- Passed into HttpConnection (TcpClient descendant) by HttpClient, using pinCertificate method
+ *  	- When certificate validated, memory is released
+ *
+ */
+struct SslFingerprints {
 	uint8_t* certSha1 = nullptr; // << certificate SHA1 fingerprint
 	uint8_t* pkSha256 = nullptr; // << public key SHA256 fingerprint
 
+	~SslFingerprints()
+	{
+		free();
+	}
+
 	void free()
 	{
-		delete certSha1;
+		delete[] certSha1;
 		certSha1 = nullptr;
-		delete pkSha256;
+		delete[] pkSha256;
 		pkSha256 = nullptr;
 	}
-} SslFingerprints;
+
+	bool setSha1(const uint8_t* cert, unsigned length)
+	{
+		return setValue(certSha1, SHA1_SIZE, cert, length);
+	}
+
+	bool setSha256(const uint8_t* cert, unsigned length)
+	{
+		return setValue(pkSha256, SHA256_SIZE, cert, length);
+	}
+
+	/** @brief Make copy of SHA1 certificate from data stored in flash
+	 *  @param cert
+	 */
+	bool setSha1_P(const uint8_t* cert, unsigned length)
+	{
+		// Word-aligned and sized buffers don't need special handling

```

Comment:
Is the word-aligning planned to be added here? If not then this method is identical to the non _P versions and my question is why is that needed?

Reply:
It's a matter of semantics; it also means we could add an optimisation internally which avoids copying the fingerprint data but instead takes a reference to it.

Reply:
> but I'd also change to a `class` and make the certificate data private, which would break existing user's code...

Let's leave this change for now and schedule it for the next version.